circleci: failed tests: TestTxnPutOutOfOrder

The following test appears to have failed:

[#17930](https://circleci.com/gh/cockroachdb/cockroach/17930):

```
I160516 23:49:07.288063 kv/dist_sender.go:986  application error: range 0: replica <nil> not leader; leader is <nil>
I160516 23:49:07.288188 kv/dist_sender.go:986  application error: range 0: replica <nil> not leader; leader is <nil>
I160516 23:49:07.288312 kv/dist_sender.go:986  application error: range 0: replica <nil> not leader; leader is <nil>
I160516 23:49:07.288445 kv/dist_sender.go:986  application error: range 0: replica <nil> not leader; leader is <nil>
I160516 23:49:07.288570 kv/dist_sender.go:986  application error: range 0: replica <nil> not leader; leader is <nil>
panic: test timed out after 1m10s

goroutine 14593 [running]:
panic(0x154d6a0, 0xc8217c4120)
	/usr/local/go/src/runtime/panic.go:464 +0x3e6
testing.startAlarm.func1()
	/usr/local/go/src/testing/testing.go:725 +0x14b
created by time.goFunc
	/usr/local/go/src/time/sleep.go:129 +0x3a

goroutine 1 [chan receive]:
testing.RunTests(0x1ce8ad0, 0x246a220, 0x10f, 0x10f, 0xc8202ca401)
	/usr/local/go/src/testing/testing.go:583 +0x8d2
testing.(*M).Run(0xc820209ef8, 0xc8202c9c98)
	/usr/local/go/src/testing/testing.go:515 +0x81
github.com/cockroachdb/cockroach/storage_test.TestMain(0xc820209ef8)
	/go/src/github.com/cockroachdb/cockroach/storage/main_test.go:54 +0x1de
main.main()
	github.com/cockroachdb/cockroach/storage/_test/_testmain.go:604 +0x114

goroutine 17 [syscall, 1 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1998 +0x1

goroutine 20 [chan receive]:
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x2741c20)
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1011 +0x64
created by github.com/cockroachdb/cockroach/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:598 +0x8a

goroutine 14511 [semacquire]:
sync.runtime_Syncsemacquire(0xc8208bd010)
	/usr/local/go/src/runtime/sema.go:241 +0x201
sync.(*Cond).Wait(0xc8208bd000)
	/usr/local/go/src/sync/cond.go:63 +0x9b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).Quiesce(0xc821720fc0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:354 +0x1eb
github.com/cockroachdb/cockroach/util/stop.(*Stopper).Stop(0xc821720fc0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:300 +0x1a7
github.com/cockroachdb/cockroach/storage_test.TestTxnPutOutOfOrder(0xc821654990)
	/go/src/github.com/cockroachdb/cockroach/storage/client_replica_test.go:322 +0xdaa
testing.tRunner(0xc821654990, 0x246b990)
	/usr/local/go/src/testing/testing.go:473 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:582 +0x892

goroutine 14597 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:356 +0x303
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc821720fc0, 0xc821b2cdc0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 14601 [select]:
github.com/cockroachdb/cockroach/storage.(*idAllocator).start.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/id_alloc.go:123 +0x86a
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc821720fc0, 0xc820f4bbd0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 14584 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:992 +0x3cf
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc821720fc0, 0xc8217f4b20)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 14598 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:356 +0x303
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc821720fc0, 0xc821b2cde0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 14583 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).Start.func4()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:921 +0x13d
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc821720fc0, 0xc8217f4ae0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 14599 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:356 +0x303
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc821720fc0, 0xc821b2ce20)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 14588 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:356 +0x303
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc821720fc0, 0xc8217f4b80)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 14585 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func2()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1011 +0x3cf
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc821720fc0, 0xc8217f4b40)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 14579 [select]:
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).startStats(0xc821750060)
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:203 +0xd21
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).(github.com/cockroachdb/cockroach/kv.startStats)-fm()
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:188 +0x20
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc821720fc0, 0xc821472f90)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 14581 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).processRaft.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:2059 +0xbf1
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc821720fc0, 0xc821473f90)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 14580 [select]:
github.com/cockroachdb/cockroach/storage.(*StorePool).start.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store_pool.go:225 +0x23f
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc821720fc0, 0xc821b93440)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 14594 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:356 +0x303
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc821720fc0, 0xc821b2cd60)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 14596 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:356 +0x303
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc821720fc0, 0xc821b2cda0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 14512 [chan receive]:
github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open.func1(0xc821861a40)
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:164 +0x3a
created by github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:165 +0x8f4

goroutine 14595 [runnable]:
github.com/cockroachdb/cockroach/storage.(*rangeBTreeKey).getKey(0xc820c4f800, 0x7fb8d4eb3268, 0xc820c4f800, 0xc8211c49d8)
	<autogenerated>:245
github.com/cockroachdb/cockroach/storage.(*Replica).Less(0xc821635700, 0x7fb8d4eb3268, 0xc820c4f800, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:182 +0x9a
github.com/google/btree.(*BTree).AscendGreaterOrEqual.func1(0x7fb8d4eb3290, 0xc821635700, 0x3)
	/go/src/github.com/google/btree/btree.go:591 +0x46
github.com/google/btree.(*node).iterate(0xc821826340, 0xc8211c4b20, 0x1ce85e0, 0xc8211c4be0, 0xc820c4f800)
	/go/src/github.com/google/btree/btree.go:427 +0xb5
github.com/google/btree.(*BTree).AscendGreaterOrEqual(0xc821b935e0, 0x7fb8d4eb3268, 0xc820c4f800, 0xc8211c4be0)
	/go/src/github.com/google/btree/btree.go:593 +0x71
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc821720fc0, 0xc821b2cd80)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 14589 [select]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc8215ec4d0, 0xececc5340, 0x39af515d, 0x2741040, 0xc8208bcfc0, 0xc821720fc0, 0xc821635700, 0xc820439e98)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:161 +0x568
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1.1(0xc821635700, 0xc82171d4a0)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:203 +0x78
github.com/cockroachdb/cockroach/storage.(*storeRangeSet).Visit(0xc82171d860, 0xc821826480)
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:231 +0x246
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:205 +0x21e
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc821720fc0, 0xc8217f4be0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62
FAIL	github.com/cockroachdb/cockroach/storage	70.060s
=== RUN   TestBatchBasics
I160516 23:48:00.805555 storage/engine/rocksdb.go:137  opening in memory rocksdb instance
I160516 23:48:00.807177 storage/engine/rocksdb.go:178  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.01s)
=== RUN   TestBatchRepr
I160516 23:48:00.812839 storage/engine/rocksdb.go:137  opening in memory rocksdb instance
I160516 23:48:00.813746 storage/engine/rocksdb.go:178  closing in-memory rocksdb instance
--- PASS: TestBatchRepr (0.01s)
=== RUN   TestBatchGet
I160516 23:48:00.816294 storage/engine/rocksdb.go:137  opening in memory rocksdb instance

```
Please assign, take a look and update the issue accordingly.

--- a/kv/dist_sender_test.go
+++ b/kv/dist_sender_test.go
@@ -1747,3 +1747,24 @@ func TestSlowLeaderRetry(t *testing.T) {
 		t.Fatal(pErr)
 	}
 }
+
+func TestSenderTransport(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	transport, err := SenderTransportFactory(
+		tracing.NewTracer(),
+		client.SenderFunc(
+			func(
+				_ context.Context, _ roachpb.BatchRequest,
+			) (r *roachpb.BatchResponse, e *roachpb.Error) {
+				return
+			},
+		))(SendOptions{}, &rpc.Context{}, nil, roachpb.BatchRequest{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	transport.SendNext(make(chan BatchCall, 1))
+	if !transport.IsExhausted() {
+		t.Fatalf("transport is not exhausted")
+	}
+	transport.Close()
+}

--- a/kv/transport.go
+++ b/kv/transport.go
@@ -283,6 +283,10 @@ func (s *senderTransport) IsExhausted() bool {
 }
 
 func (s *senderTransport) SendNext(done chan BatchCall) {
+	if s.called {
+		panic("called an exhausted transport")
+	}
+	s.called = true
 	sp := s.tracer.StartSpan("node")
 	defer sp.Finish()
 	ctx := opentracing.ContextWithSpan(context.Background(), sp)

